src/main/java/com/fortify/plugin/jenkins/steps/FortifyStep.java

```diff
@@ -53,42 +54,57 @@ protected void setLastBuild(Run<?, ?> lastBuild) {
 	}
 
 	/**
-	 * Search for the executable filename in FORTIFY_HOME or PATH environment
-	 * variables or workspace
-	 * 
+	 * Search for the executable filename in executable home directory or on PATH environment
+	 * variable or in workspace
+	 *
 	 * @param filename
-	 * @param checkFortifyHome
 	 * @param build
 	 * @param workspace
-	 * @param launcher
 	 * @param listener
-	 * @param msg
+	 * @param targetEnvVarName
 	 * @return found executable
 	 * @throws InterruptedException
 	 * @throws IOException
 	 */
-	protected String getExecutable(String filename, boolean checkFortifyHome, Run<?, ?> build, FilePath workspace,
-			Launcher launcher, TaskListener listener, String msg) throws InterruptedException, IOException {
+	protected String getExecutable(String filename, Run<?, ?> build, FilePath workspace,
+								   TaskListener listener, String targetEnvVarName) throws InterruptedException, IOException {
+		PrintStream logger = listener.getLogger();
 		EnvVars env = build.getEnvironment(listener);
-		String fortifyHome = null;
+
+		String home = null;
 		String path = null;
+		boolean isEnvVarSetProperly = true;
 		// check env variables defined in Jenkins master
 		for (Map.Entry<String, String> entry : env.entrySet()) {
-			String key = entry.getKey();
-			if ("FORTIFY_HOME".equals(key)) {
-				if (checkFortifyHome) {
-					fortifyHome = entry.getValue();
+			String envVarName = entry.getKey();
+			String envVarValue = entry.getValue();
+			if (targetEnvVarName != null && targetEnvVarName.equals(envVarName)) {
+				home = envVarValue;
+				if (envVarValue.endsWith("bin") || envVarValue.endsWith("bin/") || envVarValue.endsWith("bin\\")) {
```

Comment:
Suggestion:
Create helper method.
`if (endsWithBin(envVarValue)) {
`
```
private static boolean endsWithBin(String str) {
    return str.endsWith("bin") || str.endsWith("bin/") || str.endsWith("bin\\");
}
```

Reply:
Thanks for the tip. Added the fix.